### ISSUE TITLE
inspect.signature() raises RuntimeError on failed to resolve the default argument value

### ISSUE THREAD

**Issue by b922c0d2-75a2-4203-8f93-21a8f59b93af** [python/cpython#87284]
BPO | [43118](https://bugs.python.org/issue43118)
--- | :---
Nosy | @gvanrossum, @hongweipeng, @miss-islington, @tk0miya
PRs | <li>python/cpython#30285</li><li>python/cpython#30765</li><li>python/cpython#30766</li>
Files | <li>[inspect.patch](https://bugs.python.org/file49788/inspect.patch "Uploaded as text/plain at 2021-02-03.15:04:11 by @tk0miya")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-02-03.15:04:12.206>
labels = ['type-bug', 'library', '3.9', '3.10', '3.11']
title = 'inspect.signature() raises RuntimeError on failed to resolve the default argument value'
updated_at = <Date 2022-01-22.17:28:57.262>
user = 'https://github.com/tk0miya'
```

bugs.python.org fields:
```python
activity = <Date 2022-01-22.17:28:57.262>
actor = 'gvanrossum'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2021-02-03.15:04:12.206>
creator = 'tkomiya'
dependencies = []
files = ['49788']
hgrepos = []
issue_num = 43118
keywords = ['patch']
message_count = 5.0
messages = ['386212', '386213', '411188', '411196', '411268']
nosy_count = 4.0
nosy_names = ['gvanrossum', 'hongweipeng', 'miss-islington', 'tkomiya']
pr_nums = ['30285', '30765', '30766']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue43118'
versions = ['Python 3.9', 'Python 3.10', 'Python 3.11']
```

</p></details>

**Comment by alittlesir** [comment]
any question about this issue? if not, i think it is better to close it

**Comment by gvanrossum** [comment]
Looks like this was fixed in 3.9, the repro doesn't raise there.
